custom_components/deltadore_tydom/ha_entities.py

```diff
@@ -895,6 +895,42 @@ def is_closed(self) -> bool:
             LOGGER.error("Unknown state for device %s", self._device.device_id)
             return True
 
+class HaWindowBinary(BinarySensorEntity, HAEntity):
```

Comment:
At first glance, the code of this class is the same as in `HADoorBinary` modulo the `_attr_device_class` attribute, it would be nice to factorize this code.